kv/kvserver/reports: TestConstraintConformanceReportIntegration failed

kv/kvserver/reports.TestConstraintConformanceReportIntegration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11247941?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11247941?buildTab=artifacts#/) on release-23.1 @ [daeaec2e81733e9f26205d7bab04f819778d895b](https://github.com/cockroachdb/cockroach/commits/daeaec2e81733e9f26205d7bab04f819778d895b):


```
=== RUN   TestConstraintConformanceReportIntegration
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1ce2639b2597bee9434a4c1e166a131c/logTestConstraintConformanceReportIntegration3824425511
    test_log_scope.go:79: use -show-logs to present logs inline
    reporter_test.go:83: condition failed to evaluate within 45s: violation not detected yet
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1ce2639b2597bee9434a4c1e166a131c/logTestConstraintConformanceReportIntegration3824425511
--- FAIL: TestConstraintConformanceReportIntegration (88.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConstraintConformanceReportIntegration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30467

--- a/pkg/kv/kvserver/reports/reporter_test.go
+++ b/pkg/kv/kvserver/reports/reporter_test.go
@@ -48,6 +48,8 @@ func TestConstraintConformanceReportIntegration(t *testing.T) {
 	// don't make progress.
 	skip.UnderStressRace(t)
 	skip.UnderRace(t, "takes >1min under race")
+	// Similarly, skip the test under deadlock builds.
+	skip.UnderDeadlock(t, "takes >1min under deadlock")
 
 	ctx := context.Background()
 	tc := serverutils.StartNewTestCluster(t, 5, base.TestClusterArgs{

